src/components/experiment-tracking/experiment-primary-toolbar/experiment-primary-toolbar.js

```diff
@@ -1,23 +1,47 @@
 import React from 'react';
+import { useMutation } from '@apollo/client';
+import { UPDATE_RUN_DETAILS } from '../../../apollo/mutations';
 import IconButton from '../../icon-button';
 import PencilIcon from '../../icons/pencil';
+import BookmarkIcon from '../../icons/bookmark';
+import BookmarkStrokeIcon from '../../icons/bookmark-stroke';
 import PrimaryToolbar from '../../primary-toolbar';
 import ShowChangesIcon from '../../icons/show-changes';
 
 export const ExperimentPrimaryToolbar = ({
   enableComparisonView,
   enableShowChanges,
+  selectedRunData,
   setEnableShowChanges,
   setSidebarVisible,
   showChangesIconDisabled,
   showRunDetailsModal,
   sidebarVisible,
 }) => {
+  const [updateRunDetails] = useMutation(UPDATE_RUN_DETAILS);
+
+  const toggleBookmark = () => {
+    updateRunDetails({
+      variables: {
+        runId: selectedRunData.id,
+        runInput: { bookmark: !selectedRunData?.bookmark },
+      },
+    });
+  };
+
```

Comment:
This function is used in 2 places ( both here and `runMetadata.js` ) - I would extract that out to the main experiment wrapper to pass it down to both components instead.

Reply:
Good idea. Though instead of passing it as a prop I'm going to make it a custom hook so we can pass it around freely as we wish.